client/src/components/form.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from "react";
+import {
+  Typography,
+  Container,
+  CssBaseline,
+  makeStyles,
+  Button,
+  Divider,
+} from "@material-ui/core";
+
+import { Link } from "react-router-dom";
+import TextInput from "./TextInput";
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(8),
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  form: {
+    marginTop: theme.spacing(1),
+    width: "100%",
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  submit: {
+    marginTop: theme.spacing(4),
+    backgroundColor: "#69E781",
+    color: "white",
+    height: "4em",
+    width: "50%",
+  },
+  description: {
+    margin: theme.spacing(0.5, 3, 4),
+  },
+  divider: {
+    backgroundColor: "#69E781",
+    height: "0.25em",
+    width: "8vh",
+    margin: "3em",
+  },
+}));
+
+const Form = ({
```

Comment:
I'd change the name of this component and the file to something less general, as it's not a general form but only a form for Signup and Login Pages